### PR TITLE
API: Create an Event

### LINKED ISSUE #133: API: Endpoint to create an event
Create an API endpoint that Creates a new Event and make sure that the event is created along and accordingly with all the foreign keys. The endpoint should consider that only Admin role can create events.

image

The fields venue, description and max_participants should be null or using the default values, check the json examples here -  #67 

image

This API will be used on this component - https://github.com/HackYourFuture-CPH/web-dev-rooms/issues/132

The endpoint has to be documented on Swagger

### PR DESCRIPTION
**Pull request by Deepti-Dk**
# Description

This PR is to create an event with FK constraints working. I have updated existing 'events.router.js' and 'events.controller.js' with POST to create an event.

Fixes #133 

# How to test?

tested on swagger and postman

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/routes/api-router.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ router.use('/topics', TopicsRouter);
 router.use('/skills', skillsRouter);
 
 router.use('/events', eventsRouter);
+
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:49]: Undo change in this file.

**Review by tpetrina — CHANGES_REQUESTED** [review]
The backend code looks neat and tidy, but there are other files in the frontend and I don't know what to do with them? Can you please separate them into another PR so that this one goes in smoothly?

File: src/server/api/routes/api-router.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ const groupsRouter = require('./groups.router');
 
 const slackRouter = require('./slack.router');
 
+// const createEventRouter = require('./create-event.router');
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:18]: Remove commented code.

File: src/server/api/routes/api-router.js [hunk 3]
```diff
@@ -50,4 +52,9 @@ router.use('/groups', groupsRouter);
 
 router.use('/signin', slackRouter);
 
+// router.use('/createEvent', createEventRouter);
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:55]: Remove commented code.

File: src/server/api/routes/api-router.js [hunk 4]
```diff
@@ -50,4 +52,9 @@ router.use('/groups', groupsRouter);
...
+router.use('/', function (req, res) {
+  res.redirect(`${process.env.API_PATH}/documentation`);
+});
+
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js:59]: Why is this here now?

**Review by tpetrina — APPROVED** [review]
Please cleanup before you merge.

File: src/server/api/routes/api-router.js (hunk 2 above)
Comment (Deepti-Dk) [reply src/server/api/routes/api-router.js:18]: Done

File: src/server/api/routes/api-router.js (hunk 4 above)
Comment (Deepti-Dk) [reply src/server/api/routes/api-router.js]: deleted it.

File: src/server/api/routes/api-router.js (hunk 3 above)
Comment (Deepti-Dk) [reply src/server/api/routes/api-router.js]: done

**Review by yukafjaponesa — APPROVED** [review]
looks nice.

File: src/server/api/controllers/events.controller.js [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
+    venue: body.venue,
+    description: body.description,
+    created_by: body.createdBy,
+  });
+
+  return {
+    successful: true,
+  };
+};
+
```
Comment (AkoAyukngat) [inline src/server/api/controllers/events.controller.js:16]: this is smooth and clean :)

File: src/server/api/routes/events.router.js [hunk 6]
```diff
@@ -7,6 +7,51 @@ const router = express.Router({ mergeParams: true });
...
+ *              type: string
+ *            createdBy:
+ *              type: integer
+ *    responses:
+ *      201:
+ *        description: Event created
+ *      5XX:
+ *        description: Unexpected error.
+ */
+
```
Comment (AkoAyukngat) [inline src/server/api/routes/events.router.js:44]: you well understood what you were doing.

File: src/server/api/controllers/events.controller.js [hunk 7]
```diff
@@ -1,6 +1,19 @@
...
 const Error = require('../lib/utils/http-error');
 
+const createEvent = async (body) => {
+  await knex('events').insert({
+    event_type: body.eventType,
+    venue: body.venue,
+    description: body.description,
+    created_by: body.createdBy,
+  });
+
```
Comment (AkoAyukngat) [inline src/server/api/controllers/events.controller.js:11]: this is smooth and clean :)

**Review by tpetrina — APPROVED** [review]

**Pull request merged**